__tests__/Ship.tests.js

```diff
@@ -0,0 +1,39 @@
+const Ship = require('../src/Ship.js');
+
+describe('Ship', () => {
+  let dover;
+  let calais;
+  let itinerary;
+  let ship;
+  const port = {
+    removeShip: jest.fn(),
+    addShip: jest.fn(),
+  };
+  beforeEach(() => {
+    dover = { ...port, name: 'Dover', ships: [] };
+    calais = { ...port, name: 'Calais', ships: [] };
+    itinerary = { ports: [dover, calais] };
+    ship = new Ship(itinerary);
+  });
+  it('A new ship can be instantiated', () => {
+    expect(ship).toBeInstanceOf(Object);
+  });
+  it('has a starting port', () => {
+    expect(ship.currentPort.name).toBe('Dover');
+  });
+  it('Ship can set sail', () => {
+    ship.setSail();
+    expect(ship.currentPort).toBeFalsy();
+    expect(dover.ships).not.toContain(ship);
```

Comment:
with tests containing `.not` keyword, I like to exercise what was the the state of things before action took place.

E.g:

```
    expect(dover.ships).toContain(ship);
    ship.setSail(); // now I can be sure, calling this method indeed is the reason why my tests pass!
    expect(ship.currentPort).toBeFalsy();
    expect(dover.ships).not.toContain(ship);
```